packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1430,22 +1430,97 @@ export class TransactionController extends BaseController<
     return rawTransactions;
   }
 
+  /**
+   * Updates the transaction status associated with a specific transaction ID.
+   *
+   * @param transactionId - The ID of the transaction to update.
+   * @param options - The custodial transaction options to update.
+   * @param options.custodyStatus - The new custody status value to be assigned.
+   * @param options.errorMessage - The error message to be assigned in case transaction status update to failed.
+   * @param options.hash - The new hash value to be assigned.
+   * @param options.status - The new status value to be assigned.
+   */
+  updateCustodialTransaction(
+    transactionId: string,
+    {
+      custodyStatus,
+      errorMessage,
+      hash,
+      status,
+    }: {
+      custodyStatus?: string;
+      errorMessage?: string;
+      hash?: string;
+      status?: TransactionStatus;
+    },
+  ) {
+    let transactionMeta;
+    transactionMeta = this.getTransaction(transactionId);
+    const noteDetails = [];
+
+    if (!transactionMeta) {
+      throw new Error(`Cannot update status as no transaction metadata found.`);
+    }
+
+    if (!transactionMeta.custodyId) {
+      throw new Error('Transaction is not custodian.');
+    }
+    if (status === TransactionStatus.signed) {
+      transactionMeta.status = status;
+      noteDetails.push(`status to ${status}`);
+    }
+
+    if (status === TransactionStatus.submitted) {
+      transactionMeta.submittedTime = new Date().getTime();
+      transactionMeta.status = status;
+      noteDetails.push(`status to ${status}`);
+    }
+
+    if (status === TransactionStatus.failed) {
+      transactionMeta = {
+        ...transactionMeta,
+        error: normalizeTxError(new Error(errorMessage)),
+        status: TransactionStatus.failed,
+      };
+      noteDetails.push(`status to ${status}`);
+    }
+
+    if (custodyStatus) {
+      transactionMeta.custodyStatus = custodyStatus;
+      noteDetails.push(`custody status`);
```

Comment:
The history already provides a full diff of the state over the time so would it be cleaner to keep the note at a high level such as `TransactionController:updateCustodialTransaction - Custodial transaction updated`?

Reply:
yeap I'm in favor of simplifying it. Pushed the changes to keep the note at a high level.